packages/babel-parser/src/tokenizer/index.js

```diff
@@ -438,6 +438,9 @@ export default class Tokenizer extends ParserErrors {
         this.finishToken(tt.bracketHashL);
       }
       this.state.pos += 2;
+    } else if (isIdentifierStart(next) || next === charCodes.backslash) {
```

Comment:
Can you explain the `backslash`? I'm confused why `#\` would be valid.

Reply:
It's for the escaped private names: `#\u0061`.